packages/rust/server/src/models/member.rs

```diff
@@ -46,23 +47,62 @@ pub struct GroupIDs {
 
 impl From<&PgRow> for Member {
     fn from(row: &PgRow) -> Self {
+        let row_id_in: i32 = row.get("role_id");
         Member {
             id: row.get("id"),
             user_id: row.get("user_id"),
             group_id: row.get("group_id"),
-            role_id: Role::try_from((row.get::<f32, &str>("role_id")) as u8).unwrap(),
+            role_id: Role::try_from(row_id_in as u8).unwrap(),
             created: row.get("created"),
         }
     }
 }
 
 impl Member {
-    //TODO check if admin and if group is empty
     pub async fn delete_all_by_user_id(db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
-        sqlx::query("DELETE FROM members WHERE user_id = $1")
+        let groups = sqlx::query("SELECT group_id FROM member WHERE user_id = $1")
+            .bind(user_id)
+            .fetch_all(db)
+            .await?;
+
+        //checking for edge cases
+        let members_size_len = groups.len();
+        for i in groups {
+            let group_id: i32 = i.get("group_id");
+
+            //cannot delete group if user to delete is only admin
+            let admin_size =
+                sqlx::query("SELECT user_id FROM member WHERE group_id = $1 AND role_id = $2")
+                    .bind(group_id)
+                    .bind(Role::Admin as i32)
+                    .fetch_all(db)
+                    .await?;
+            if admin_size.len() == 1 {
+                let admin_id: i32 = admin_size[0].get("user_id");
+                if admin_id == user_id && members_size_len != 1 {
+                    //TODO Preferable to return an error from routes, not models, that is not sqlx and includes the id of the groups in quesetion
+                    return Err(sqlx::Error::Protocol(
+                        "There must be at least one Admin in the group.".to_string(),
```

Comment:
yeah no

Reply:
I believe this has been fixed, could you check?